src/java/org/apache/cassandra/locator/AlibabaCloudSnitch.java

```diff
@@ -17,130 +17,47 @@
-        splits = az.split("-");
-        ecsZone = splits[splits.length - 1];
-
-        int lastRegionIndex = az.lastIndexOf("-");
-        ecsRegion = az.substring(0, lastRegionIndex);
+    private static final String DEFAULT_METADATA_SERVICE_URL = "http://100.100.100.200";
+    private static final String ZONE_NAME_QUERY_URL = "/latest/meta-data/zone-id";
 
-        String datacenterSuffix = (new SnitchProperties()).get("dc_suffix", "");
-        ecsRegion = ecsRegion.concat(datacenterSuffix);
-        logger.info("AlibabaSnitch using region: {}, zone: {}.", ecsRegion, ecsZone);
-    
-    }
-    
-    String alibabaApiCall(String url) throws ConfigurationException, IOException, SocketTimeoutException
+    public AlibabaCloudSnitch() throws IOException
     {
-        // Populate the region and zone by introspection, fail if 404 on metadata
-        HttpURLConnection conn = (HttpURLConnection) new URL(url).openConnection();
-        DataInputStream d = null;
-        try
-        {
-            conn.setConnectTimeout(HTTP_CONNECT_TIMEOUT);
-            conn.setRequestMethod("GET");
-            
-            int code = conn.getResponseCode();
-            if (code != HttpURLConnection.HTTP_OK)
-                throw new ConfigurationException("AlibabaSnitch was unable to execute the API call. Not an ecs node? and the returun code is " + code);
-
-            // Read the information. I wish I could say (String) conn.getContent() here...
-            int cl = conn.getContentLength();
-            byte[] b = new byte[cl];
-            d = new DataInputStream((FilterInputStream) conn.getContent());
-            d.readFully(b);
-            return new String(b, StandardCharsets.UTF_8);
-        }
-        catch (SocketTimeoutException e)
-        {
-            throw new SocketTimeoutException("Timeout occurred reading a response from the Alibaba ECS metadata");
-        }
-        finally
-        {
-            FileUtils.close(d);
-            conn.disconnect();
-        }
+        this(new SnitchProperties());
     }
-    
-    @Override
-    public String getRack(InetAddressAndPort endpoint)
+
+    public AlibabaCloudSnitch(SnitchProperties properties) throws IOException
     {
-        if (endpoint.equals(FBUtilities.getBroadcastAddressAndPort()))
-            return ecsZone;
-        EndpointState state = Gossiper.instance.getEndpointStateForEndpoint(endpoint);
-        if (state == null || state.getApplicationState(ApplicationState.RACK) == null)
-        {
-            if (savedEndpoints == null)
-                savedEndpoints = SystemKeyspace.loadDcRackInfo();
-            if (savedEndpoints.containsKey(endpoint))
-                return savedEndpoints.get(endpoint).get("rack");
-            return DEFAULT_RACK;
-        }
-        return state.getApplicationState(ApplicationState.RACK).value;
-    
+        this(properties, new DefaultCloudMetadataServiceConnector(properties.get(METADATA_URL_PROPERTY,
```

Comment:
what about merge METADATA_URL_PROPERTY and ZONE_NAME_QUERY_URL  together as before not just split them into two Strings and make them configurable. 
or make ZONE_NAME_QUERY_URL configurable through Snitchpropetry ? (In this way I may suggest changing the name to ZONE_NAME_QUERY_URL_PATH)

Reply:
See above comment about why it is split into two properties.